src/App.js

```diff
@@ -1,75 +1,140 @@
 import React from 'react';
+import TodoList from './components/TodoList/TodoList';
+import FormTodo from './components/FormTodo/FormTodo';
+import Footer from './components/Footer/Footer';
 
-function App() {
-  return (
-    <section className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
+class App extends React.Component {
+  state = {
+    todos: [],
+    originalTodos: [],
+    activeTab: '',
+    statusAllTodos: true,
+  };
 
-        <input
-          className="new-todo"
-          placeholder="What needs to be done?"
-        />
-      </header>
+  addTodo = (todo) => {
+    this.setState(prevState => ({
+      todos: [...prevState.todos, todo],
+      originalTodos: [...prevState.originalTodos, todo],
+    }));
 
-      <section className="main" style={{ display: 'block' }}>
-        <input type="checkbox" id="toggle-all" className="toggle-all" />
-        <label htmlFor="toggle-all">Mark all as complete</label>
+    switch (this.state.activeTab) {
+      case 'active':
+        this.filterActive();
+        break;
+      case 'all':
+        this.filterAll();
+        break;
+      case 'completed':
+        this.filterCompleted();
+        break;
+      default:
+    }
+  };
 
-        <ul className="todo-list">
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-1" />
-              <label htmlFor="todo-1">sdfsdfsdf</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
+  handleDelete = (todo) => {
+    !todo.comleted
+      && this.setState(prevState => ({ items: prevState.items - 1 }));
+    this.setState(prevState => ({
+      todos: [...prevState.todos].filter(elem => elem.id !== todo.id),
+      originalTodos: [...prevState.todos].filter(elem => elem.id !== todo.id),
+    }));
+  };
 
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-2" />
-              <label htmlFor="todo-2">sakgjdfgkhjasgdhjfhs</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
+  handleClickCheckBox = (index) => {
+    this.setState(prevState => ({
+      todos: prevState.todos.map((elem, i) => (i === index
```

Comment:
если `todos`, почему `elem`?
```suggestion
      todos: prevState.todos.map((todo, i) => (i === index
```

Reply:
не исправлено